src/containers/NFT/NFTHeader/NFTHeader.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useContext, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useQuery } from 'react-query'
+import Loader from '../../shared/components/Loader'
+import '../../shared/css/nested-menu.scss'
+import './styles.scss'
+import SocketContext from '../../shared/SocketContext'
+import Tooltip from '../../shared/components/Tooltip'
+import { getNFTInfo } from '../../../rippled/lib/rippled'
+import { formatNFTInfo } from '../../../rippled/lib/utils'
+import {
+  analytics,
+  ANALYTIC_TYPES,
+  BAD_REQUEST,
+  HASH_REGEX,
+} from '../../shared/utils'
+import Details from './Details'
+import Settings from './Settings'
+import Account from '../../shared/components/Account'
+
+interface Props {
+  tokenId: string
+  setError: (error: number | null) => void
+}
+
+const NFTHeader = (props: Props) => {
+  const { t } = useTranslation()
+  const { tokenId, setError } = props
+  const rippledSocket = useContext(SocketContext)
+  const [tooltip, setTooltip] = useState(null)
+  const { data: rawData, isFetching: loading } = useQuery(
+    ['getNFTInfo'],
+    async () => getNFTInfo(rippledSocket, tokenId),
```

Comment:
why is this async? does useQuery require that?

Reply:
I don't think it requires it.  Just that a promise is returned and getNFTInfo does that.